pyspeckit/spectrum/units.py

```diff
@@ -417,6 +423,11 @@ def __new__(self, xarr, unit="Hz", frame='rest', frame_offset=0.0,
         else:
             subarr.velocity_convention = velocity_convention
 
+        subarr._equivalencies = equivalencies
```

Comment:
`subarr._equivalencies` should be set by the `find_equivalencies` method that you've written